arrow-select/src/filter.rs

```diff
@@ -250,6 +243,20 @@ impl FilterBuilder {
         self
     }
 
+    /// Determines if calling [FilterBuilder::optimize] is beneficial for the given type even when
+    /// filtering just a single array.
```

Comment:
I think it would help to add a little more context here

```suggestion
    /// Determines if calling [FilterBuilder::optimize] is beneficial for the given type even when
    /// filtering just a single array.
    ///
    /// In general optimizing will be beneficial if there is more than one child array (e.g. a StructArray with multiple fields)